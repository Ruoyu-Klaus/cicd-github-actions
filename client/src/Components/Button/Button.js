/*
 * @Author: Ruoyu
 * @FilePath: /cicd-github-actions/client/src/Components/Button/Button.js
 */
import React from 'react';
import styles from './Button.module.css';
import classnames from 'classnames/bind';
let cx = classnames.bind(styles);

function Button(props) {
  const { size, type, children } = props;
  let className = cx({
    base: true,
    primary: type ? type === 'primary' : true,
    secondary: type === 'secondary',
    large: size === 'large',
    middle: size ? size === 'middle' : true,
    small: size === 'small',
  });

  return <button className={className}>{children}</button>;
}

export default Button;
