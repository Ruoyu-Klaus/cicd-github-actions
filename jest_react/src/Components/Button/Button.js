/*
 * @Author: Ruoyu
 * @FilePath: /get-start-tdd/jest_react/src/Components/Button/Button.js
 */
import React from 'react';
import styles from './Button.module.css';
import classnames from 'classnames/bind';
let cx = classnames.bind(styles);

function Button(props) {
  const { size, type, children } = props;
  let className = cx({
    base: true,
    primary: type ? type === 'primary' : true,
    secondary: type === 'secondary',
    large: size === 'large',
    middle: size ? size === 'middle' : true,
    small: size === 'small',
  });

  return <button className={className}>{children || 'Button'}</button>;
}

export default Button;
