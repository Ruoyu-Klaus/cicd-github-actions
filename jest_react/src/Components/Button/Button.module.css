.base{
  border:none;
}

.primary{
  background:#1890ff;
  color:#fff
}
.secondary{
  background:#ff4d4f;
}

.large{
  height: 40px;

}
.middle{
  height: 30px;
}
.small{
  height: 20px;
}
