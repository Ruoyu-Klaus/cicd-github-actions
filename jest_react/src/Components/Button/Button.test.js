/*
 * @Author: Ruoyu
 * @FilePath: /get-start-tdd/jest_react/src/Components/Button/Button.test.js
 */
import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

it('Button with default placeholder', () => {
  // const component = renderer.create(<Button>Button </Button>);
  // let tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
  render(<Button />);
  const btnElement = screen.getByText(/button/i);
  expect(btnElement).toBeInTheDocument();
});
