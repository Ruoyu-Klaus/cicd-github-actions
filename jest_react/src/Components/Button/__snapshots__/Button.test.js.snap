// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button with default placeholder 1`] = `
<button
  className="base primary middle"
>
  Button Test
</button>
`;
